src/runtime/process.rs

```diff
@@ -0,0 +1,40 @@
+use std::{ffi::OsStr, process::Stdio};
+
+#[cfg(feature = "async-std-runtime")]
+use async_std::process::{Child, Command};
+#[cfg(feature = "tokio-runtime")]
+use tokio::process::{Child, Command};
+
+use crate::error::Result;
+
+#[derive(Debug)]
+pub(crate) struct Process {
+    child: Child,
```

Comment:
DRIVERS-1861 mentions that ideally the mongocryptd process would not be a child process of the driver, since it would want to keep executing for other driver instances even after the first one exited. I looked briefly, but I'm not sure of a way to do this cleanly with `tokio`'s API. I did see some solutions that use unsafe + libc, but I assume they are not very portable. Might be worth looking into some more, but I am not super hopeful we'll find a way around this worth pursuing.

Reply:
This is complicated and messy.  My understanding of the situation:
* On Linux, a child process will continue to run if the parent process terminated normally; however, if the parent process receives `SIGINT`/`SIGTERM` that will be propagated to the children (killing them unless they have special handlers).  The normal way to avoid this is via the double-fork dance, which is nicely captured by the [`fork` crate](https://crates.io/crates/fork), but that just provides a raw `pid_t` to the parent, so we'd need to implement our own waiting/reaping logic.
* On Windows, you just need to set the [`CREATE_NEW_PROCESS_GROUP` flag](https://docs.microsoft.com/en-us/windows/win32/procthread/process-creation-flags) - this can be done in std via `CommandExt::creation_flags`, but neither Tokio nor async-std expose that in their respective `Command` APIs, and the `fork` crate doesn't support Windows.
* On OS X, it seems like pre-10.5 it was the same as Linux, but in 10.5+ it's its own thing (not entirely clear exactly what).  It looks like the `fork` crate handles this.

My inclination is to punt on this; if the user needs a persistent mongocryptd they can spawn one themselves and tell the driver to use it and not autospawn.